src/http_proxy_io.c

```diff
@@ -918,6 +922,30 @@ static int http_proxy_io_set_option(CONCRETE_IO_HANDLE http_proxy_io, const char
                 result = 0;
             }
         }
+        else if (strcmp(option_name, OPTION_USE_TLS_IO) == 0)
+        {
+            (void)xio_close(http_proxy_io_instance->underlying_io, NULL, NULL);
+            xio_destroy(http_proxy_io_instance->underlying_io);
+            http_proxy_io_instance->use_tls_io = true;
+            const IO_INTERFACE_DESCRIPTION* underlying_io_interface;
+            underlying_io_interface = platform_get_default_tlsio();
+
+            TLSIO_CONFIG tls_io_config;
+            tls_io_config.hostname = http_proxy_io_instance->proxy_hostname;
+            tls_io_config.port = http_proxy_io_instance->proxy_port;
+            http_proxy_io_instance->underlying_io = xio_create(underlying_io_interface, &tls_io_config);
```

Comment:
is the previous call xio_create in http_proxy_io_create going to fail on proxy with tls mutual auth failure?

Reply:
`http_proxy_io_create` won't fail as it's creating the socket_io instance.

it's the `socket_dowork` that will fail, for example on windows we will get error:
> Socketio_Failure: Receiving data from endpoint: 10054.

which is raised from the [socketio_win32 do_work code](https://github.com/Azure/azure-c-shared-utility/blob/fe841e6e673c6ec0c91cb45c634b3af8dab9bdd5/adapters/socketio_win32.c#L690-L695):
```c
int last_error = WSAGetLastError();
if (last_error != WSAEWOULDBLOCK && last_error != ERROR_SUCCESS)
{
    LogError("Socketio_Failure: Receiving data from endpoint: %d.", last_error);
    indicate_error(socket_io_instance);
}
```